src/routes/(app)/person/[id]/+page.svelte

```diff
@@ -31,33 +34,64 @@
     return unsubscribe;
   });
 
-  $: {
-    (async () => {
-      try {
-        person = undefined;
-        pageError = undefined;
-        if (!personId) {
-          return;
-        }
-        const data = await getPerson(personId);
-        person = data;
-      } catch (err: any) {
-        person = undefined;
-        pageError = err;
+  $: if (personId) {
+    fetchPersonData();
+  }
+
+  $: if (sortOption && cachedCredits) {
+    sortedCredits = getSortedCredits(cachedCredits, sortOption);
+  }
+
+  async function fetchPersonData() {
+    try {
+      person = undefined;
+      pageError = undefined;
+      if (!personId) {
+        return;
       }
-    })();
+      person = await getPerson(personId);
+      cachedCredits = await getPersonCredits();
+      sortedCredits = getSortedCredits(cachedCredits, sortOption);
```

Comment:
If we just have the one array of credits, `getSortedCredits` could become `sortCredits`, which would simply sort the one credits array (then also wouldn't have to return anything)